src/execution_tree/primitives/store_operation.cpp

```diff
@@ -76,7 +76,7 @@ namespace phylanx { namespace execution_tree { namespace primitives
                         "two operands"));
         }
 
-        if (!valid(operands_[0]) || !valid(operands_[1]))
+        if (!valid(operands_[0])) 
```

Comment:
```suggestion
        if (!valid(operands_[0]) || is_implicit_nil(operands[1])) 
```
This would still throw if for some reason the right hand side is invalid, but would allow for assigning explicit `None`.